### PR TITLE
Added interactive SCCharts and relative constraints

### PR DESCRIPTION
**Pull request by soerendomroes**

### CODE REVIEW

File: packages/klighd-core/src/skgraph-models.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export const LABEL_TYPE = 'label'
  * Represents the Sprotty version of its java counterpart in KLighD.
  */
 export class SKNode extends KNode {
+    // TODO this might be wrong, interactiveLayout is enabled on the parent (I think)
```
Comment (Eddykasp) [inline packages/klighd-core/src/skgraph-models.ts:41]: There's a lot of uncertainty in this comment.

File: packages/klighd-interactive/src/interactive-view.tsx [hunk 2]
```diff
@@ -18,20 +18,29 @@
 import { VNode } from "snabbdom";
 import { svg } from 'sprotty'; // eslint-disable-line @typescript-eslint/no-unused-vars
 import { KNode } from './constraint-classes';
-import { filterKNodes } from './helper-methods';
+import { filterKNodes, getSelectedNode } from './helper-methods';
 import { renderHierarchyLevel as renderHierarchyLevelLayered, renderLayeredConstraint } from './layered/layered-interactive-view';
 import { renderHierarchyLevel as renderHierarchyLevelRectPacking, renderRectPackConstraint } from './rect-packing/rect-packing-interactive-view';
+import { forbiddenNodes, renderPosIndicators, renderSetRelConstraint } from './layered/layered-relCons-view';
```
Comment (Eddykasp) [inline packages/klighd-interactive/src/interactive-view.tsx:24]: ordering of imports?

File: packages/klighd-interactive/src/klighd-interactive-mouselistener.ts [hunk 3]
```diff
@@ -44,6 +44,8 @@ export class KlighdInteractiveMouseListener extends MoveMouseListener {
      */
     private target: KNode | undefined
 
+    public relCons: boolean
```
Comment (Eddykasp) [inline packages/klighd-interactive/src/klighd-interactive-mouselistener.ts:47]: document public atttributes

File: packages/klighd-interactive/src/layered/relativeConstraint-utils.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+    return chainNodes
+}
+
+/**
+ * Determines whether a rel cons can be set between {@code node1} and {@code node2}
+ * @param node1 One of the nodes
+ * @param node2 The other one of the nodes
+ */
+export function forbiddenRC(node1: KNode, node2: KNode): boolean {
```
Comment (Eddykasp) [inline packages/klighd-interactive/src/layered/relativeConstraint-utils.ts:179]: consider refactoring the function name to something more clear

File: packages/klighd-core/src/skgraph-models.ts (hunk 1 above)
Comment (soerendomroes) [reply packages/klighd-core/src/skgraph-models.ts:41]: It is correct, the comment should be removed

[the remark this answers is not in the document]
Comment (soerendomroes) [reply packages/klighd-core/src/skgraph-models.ts:41]: Yes, it is correct and the comment should be removed

File: packages/klighd-core/src/views.tsx [hunk 5]
```diff
@@ -198,6 +210,7 @@ export class KNodeView implements IView {
         if (interactiveConstraints) {
             result.push(interactiveConstraints)
         }
+        // FIXME
```
[the remark this answers is not in the document]
Comment (soerendomroes) [reply packages/klighd-core/src/views.tsx:213]: I don't know, I guess I thought something was unclear here, I should remove it.

File: packages/klighd-interactive/src/layered/relativeConstraint-utils.ts (hunk 4 above)
Comment (soerendomroes) [reply packages/klighd-interactive/src/layered/relativeConstraint-utils.ts:179]: Maybe I should do that for all variables that are named something abbreviated.

File: packages/klighd-core/src/skgraph-models.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (soerendomroes) [reply packages/klighd-core/src/skgraph-models.ts]: It is not handled differently on the server.

**Comment by NiklasRentzCAU** [comment]
Are the PRs related to this ready for review now? I would soon take a look at them if they are.

**Comment by soerendomroes** [comment]
We need an ELK release for this, so the answer is no.

**Pull request closed without merging**